frontend/src/metabase/dashboard/components/ClickBehaviorSidebar/LinkOptions/LinkedEntityPicker.tsx

```diff
@@ -157,6 +178,39 @@ function LinkedEntityPicker({
     });
   }, [clickBehavior, updateSettings]);
 
+  const { data: dashboard } = useDashboardQuery({
+    enabled: isDash,
+    id: targetId,
+  });
+  const dashboardTabs = dashboard?.tabs ?? NO_DASHBOARD_TABS;
+  const defaultDashboardTabId: number | undefined = dashboardTabs[0]?.id;
```

Comment:
TS thinks `defaultDashboardTabId` is a `number` hence type is explicitly specified.